src/XdebugClient.php

```
@@ -499,4 +737,257 @@ public function __destruct()
+
+            // Return session metadata only - caller must establish fresh socket connection
+            // Do NOT set $this->connected = true here as no socket is actually restored
+
+            return $state['session_info'];
+        } catch (Throwable) {
+            // If metadata retrieval fails, remove stale state
+            $this->clearGlobalState();
+
+            return null;
+        }
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'reconnectToGlobalSession'. (Unused Code Rules)

(UnusedPrivateMethod)